backend/assignments/1/src/main/java/org/example/CryptoTradingApplication.java

```
@@ -0,0 +1,202 @@
+            latch.await();
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+
+    }
+
+
+
+    public void displayTopNCoins(int n) {
+        System.out.println("Displaying top " + n + " coins:");
+        List<Coins> topCoins = coins.stream()
+                .sorted(Comparator.comparingDouble(Coins::getPrice).reversed())
+                .limit(n)
+                .collect(Collectors.toList());
+
+        if (topCoins.isEmpty()) {
+            ConsoleLogger.infoMethod("No coins available.");
+        } else {
+            for (Coins coin : topCoins) {
+                ConsoleLogger.infoMethod("Code: " + coin.getCode() + ", Name: " + coin.getName() + ", Price: " + coin.getPrice());
+            }
+        }
+    }
+
+
+    public void displayTopNTraders(int n) {
+        System.out.println("Displaying top " + n + " traders:");
+        List<Trader> topTraders = traders.stream()
+                .sorted(Comparator.comparingDouble(Trader::getTotalProfitOrLoss).reversed())
+                .limit(n)
+                .collect(Collectors.toList());
+
+        if (topTraders.isEmpty()) {
+            ConsoleLogger.infoMethod("No traders available.");
+        } else {
+            for (Trader trader : topTraders) {
+
+                ConsoleLogger.infoMethod("FirstName: " + trader.getFirstName() + ", LastName: " + trader.getLastName() +
+                        ", Wallet Address: " + trader.getWalletAddress() +
+                        ", Total Profit/Loss: " + trader.getTotalProfitOrLoss());
+            }
+        }
+    }
+    private JsonNode loadJsonDataFromFile(String filePath) {
+        // Implement JSON loading logic using Jackson ObjectMapper
+        ObjectMapper objectMapper = new ObjectMapper();
+        try {
+            return objectMapper.readTree(new File(filePath));
+        } catch (IOException e) {
+            e.printStackTrace();
+            return null;
```

Lets not return `null` here. Lets throw an exception and catch it.